src/MongoDB.Shared/Hex.cs

```diff
@@ -0,0 +1,48 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Globalization;
+
+namespace MongoDB.Shared
+{
+    internal static class Hex
+    {
+        public static bool IsValidHexDigit(char c)
+        {
+            return
+                c >= '0' && c <= '9' ||
+                c >= 'a' && c <= 'f' ||
+                c >= 'A' && c <= 'F';
+        }
+
+        public static bool IsValidHexString(string s)
+        {
```

Comment:
Can we use something like : 
`long.TryParse(hex, System.Globalization.NumberStyles.HexNumber, null, out output);`
here?

Reply:
I don't want to do that because hex strings could be arbitrarily long.

This class may expand in the future to have a ParseBytes method to convert arbitrarily long hex strings to a byte array.

Reply:
Hex strings can be arbitrary length, but like a byte array.

This method isn't testing whether the hex string is a valid long, just whether it's a valid arbitrary length hex string.